### PR TITLE
MongoDB Atlas support in Haystack 2.0

### PR DESCRIPTION
**Pull request by ZanSara**
Part of https://github.com/deepset-ai/haystack-core-integrations/issues/143

Blocked by https://github.com/deepset-ai/haystack-core-integrations/pull/413 and let's wait for the first release before publishing the tile.

### CODE REVIEW

File: integrations/mongodb.md [hunk 1]
```diff
@@ -16,17 +16,64 @@ logo: /logos/mongodb.png
...
+And then you can initialize a `MongoDBAtlasDocumentStore` for Haystack with required configurations:
+
+```python
+from haystack_integrations.document_stores.mongodb_atlas import MongoDBAtlasDocumentStore
+
+document_store = MongoDBAtlasDocumentStore(
+    database_name="haystack_test",
+    collection_name="test_collection",
+)
+```
```
Comment (bilgeyucel) [inline integrations/mongodb.md:66]: Can you add a basic pipeline example here?

File: integrations/mongodb.md [hunk 2]
```diff
@@ -16,17 +16,64 @@ logo: /logos/mongodb.png
 toc: true
```
Comment (bilgeyucel) [inline integrations/mongodb.md:16]:
```suggestion
toc: true
version: Haystack 2.0
```
Also, let's update the links in the frontmatter. They should point to the haystack-core-integration code

File: integrations/mongodb.md [hunk 3]
```diff
@@ -16,17 +16,64 @@ logo: /logos/mongodb.png
...
+
+To use MongoDB Atlas as your data storage for your Haystack LLM pipelines, you must have a running database at MongoDB Atlas. For details, see [Get Started with Atlas](https://www.mongodb.com/docs/atlas/getting-started/).  
+
+Once your database is set, first export an environment variable called `MONGO_CONNECTION_STRING`, which should look more or less like this:
+
+```bash
+export MONGO_CONNECTION_STRING="mongodb+srv://<username>:<password>@<cluster_name>.gwkckbk.mongodb.net/?retryWrites=true&w=majority"
+```
+
+And then you can initialize a `MongoDBAtlasDocumentStore` for Haystack with required configurations:
```
Comment (bilgeyucel) [inline integrations/mongodb.md:57]: Can you add the docs links of MongoDBAtlasDocumentStore here

Comment (ZanSara) [reply integrations/mongodb.md:58]: Note: the link won't work until the docs are published.

**Review by bilgeyucel — APPROVED** [review]

**Pull request merged**